test_files/module/main0.d

```diff
@@ -0,0 +1,7 @@
+module modules.main0;
+
+public import modules.include;
```

Comment:
> Is it necessary for imports to be public if we're using the symbols as a base for adding imports instead of the includes?

Both approaches have its own points. I side with public imports because includes imports symbols as public and I want to avoid confusion from the user of translated library.

It is common pattern in C libraries that there is some main header file and some other internal header files responsible for particular modules (the main header includes sub-headers). And the user is expected to include only the main header file. And he expects that all the symbols from internal header files are available to him through that main header file. I we want to maintain that behavior I believe public imports are better.

However a CLI switch to control how the translated sub-modules are imported (let say `--public-submodules`) can be added to let the user of dstep to decide which behavior he wants.

> I'm not sure if we actually want to exactly map the C headers in D. Public imports is usually not idiomatic D, unless it's in a package module.

What do you mean by 'package module'? A module that gathers all other sub-modules?